src/main/java/rx/internal/operators/OperatorSubscribeOn.java

```diff
@@ -87,6 +87,7 @@ public void request(final long n) {
                                             // see unit test 'testSetProducerSynchronousRequest' for more context on this
                                             producer.request(n);
                                         } else {
+                                            // FIXME should subscription returned be added to the subscriber composite
```

Comment:
Now that I think about it. L57 is not necessary because the entire worker is added to the child on L42.